src/lib/translators/mongo.ts

```diff
@@ -233,8 +233,8 @@ export function _simplifyMongoPipeline(mongoSteps: Array<MongoStep>): Array<Mong
          * includes any inclusion, we do not want to merge those steps.
          */
         if (stepOperator === '$project') {
-          const excluded = Number(step['$project'][key] === 0);
-          merge = !Object.values(lastStep.$project).some(value => value === excluded);
+          const included = Boolean(step['$project'][key]);
+          merge = Object.values(lastStep.$project).every(value => value == included);
```

Comment:
I'm afraid you can't rely on `==` here. `1 == true` but `2 != true` for instance. You're probably stuck with converting both sides to booleans and use `===`. Or maybe use the `xor` operator:

```
          merge = Object.values(lastStep.$project).every(value => !(value ^ included));
```